### PR TITLE
*: drop backward compatibility with pre-nautilus versions

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
I believe there are more places we should cleanup. let's start with these.

File: src/messages/MMgrBeacon.h [hunk 1]
```diff
@@ -158,44 +153,22 @@ class MMgrBeacon final : public PaxosServiceMessage {
...
-      decode(mgr_features, p);
-    }
-    if (header.version >= 10) {
-      decode(clients, p);
-    }
+    decode(fsid, p);
+    std::set<std::string> module_name_list;
+    decode(module_name_list, p);
+    // Only need to unpack this field if we won't have the full
+    // ModuleInfo structures added in v7
```
Comment (tchaikov) [inline src/messages/MMgrBeacon.h:165]: remove this comment, please.

**Comment by neha-ojha** [comment]
@tchaikov https://jenkins.ceph.com/job/ceph-pull-requests/69540/consoleFull#-1520279365d63714d2-c8d8-41fc-a9d4-8dee30be4c32 is related

**Comment by tchaikov** [comment]
@neha-ojha thanks for your insights. could you please help review https://github.com/ceph/ceph-object-corpus/pull/13 as well?

File: src/common/cmdparse.cc [hunk 2]
```diff
@@ -150,23 +150,7 @@ dump_cmd_to_json(Formatter *f, uint64_t features, const string& cmd)
...
-        oss << std::string("--") << desckv["name"];
-        val = oss.str();
-        std::replace(val.begin(), val.end(), '_', '-');
-
-        desckv["type"] = "CephChoices";
-        desckv["strings"] = val;
-      }
-    }
-
+    assert(HAVE_FEATURE(features, SERVER_NAUTILUS));
```
Comment (liewegas) [inline src/common/cmdparse.cc:153]: Should we keep this one so that pre-nautlius CLI will still work?

Comment (tchaikov) [reply src/common/cmdparse.cc:153]: okay. i am going to drop this change. but i think, unlike the rados clients, ceph CLI should be considered part of in-cluster facility, and the storage admin should have control on its versions

Comment (tchaikov) [reply src/common/cmdparse.cc]: @liewegas dropped and repushed.

**Comment by tchaikov** [comment]
@jdurgin @neha-ojha @liewegas ping?

File: src/mon/OSDMonitor.cc [hunk 3]
```diff
@@ -11394,7 +11394,7 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
       err = 0;
       goto reply;
     }
-    ceph_assert(osdmap.require_osd_release >= ceph_release_t::luminous);
+    ceph_assert(osdmap.require_osd_release >= ceph_release_t::octopus);
```
Comment (jdurgin) [inline src/mon/OSDMonitor.cc:11397]: should we change this assert to nautilus in pacific?

**Review by jdurgin — APPROVED** [review]

File: src/mon/OSDMonitor.cc (hunk 3 above)
Comment (tchaikov) [reply src/mon/OSDMonitor.cc:11397]: yeah. we could. but this does not change the behavior of a pacific cluster. it improves the readability though.

**Comment by tchaikov** [comment]
need to run this against the "upgrade" test suite.

**Comment by tchaikov** [comment]
changelog

- reference the latest ceph-object-corpus

**Comment by tchaikov** [comment]
i have https://pulpito.ceph.com/kchai-2021-02-24_09:11:26-upgrade-wip-kefu2-testing-2021-02-24-1509-distro-basic-smithi/5911617/. filed https://tracker.ceph.com/issues/49450

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-02-24_02:29:54-rados-wip-kefu-testing-2021-02-23-2359-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-02-24_15:54:55-rados-wip-kefu-testing-2021-02-24-2229-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-02-25_04:36:04-rados-wip-kefu-testing-2021-02-24-2230-distro-basic-smithi/

failures tracked by

- https://tracker.ceph.com/issues/49484
- https://tracker.ceph.com/issues/49456

**Comment by tchaikov** [comment]
being tested at https://pulpito.ceph.com/kchai-2021-03-03_07:14:09-rados:upgrade-wip-kefu2-testing-2021-03-03-1236-distro-basic-smithi/

File: src/messages/MOSDPGInfo.h [hunk 4]
```diff
@@ -67,33 +67,13 @@ class MOSDPGInfo final : public Message {
...
-      // pretend to be vector<pair<pg_notify_t,PastIntervals>>
-      header.version = 5;
-      encode((uint32_t)pg_list.size(), payload);
-      for (auto& i : pg_list) {
-	encode(i, payload);   // this embeds a dup (ignored) PastIntervals
-	encode(i.past_intervals, payload);
-      }
-      return;
-    }
+    assert(HAVE_FEATURE(features, SERVER_OCTOPUS));
```
Comment (neha-ojha) [inline src/messages/MOSDPGInfo.h:70]: https://tracker.ceph.com/issues/49758 is related to this.

Comment (neha-ojha) [reply src/messages/MOSDPGInfo.h:70]: sorry, the one in MOSDPGNotify.h

File: src/messages/MOSDPGNotify.h [hunk 5]
```diff
@@ -61,34 +61,14 @@ class MOSDPGNotify final : public Message {
...
-      // pretend to be vector<pair<pg_notify_t,PastIntervals>>
-      header.version = 6;
-      encode((uint32_t)pg_list.size(), payload);
-      for (auto& i : pg_list) {
-	encode(i, payload);   // this embeds a dup (ignored) PastIntervals
-	encode(i.past_intervals, payload);
-      }
-      return;
-    }
+    assert(HAVE_FEATURE(features, SERVER_OCTOPUS));
```
Comment (neha-ojha) [inline src/messages/MOSDPGNotify.h:64]: https://tracker.ceph.com/issues/49758 is related to this change.

**Pull request merged**